vmm_core/virt_mshv/src/lib.rs

```diff
@@ -1141,22 +1168,23 @@ impl virt::PartitionMemoryMap for MshvPartitionInner {
         }
         let slot_to_use = slot_to_use.unwrap();
 
-        let mut flags = HV_MAP_GPA_READABLE;
+        let mut flags = 0;
         if writable {
-            flags |= HV_MAP_GPA_WRITABLE;
+            flags |= set_bits!(u8, MSHV_SET_MEM_BIT_WRITABLE);
         }
         if exec {
-            flags |= HV_MAP_GPA_EXECUTABLE;
+            flags |= set_bits!(u8, MSHV_SET_MEM_BIT_EXECUTABLE);
         }
 
         let mem_region = mshv_user_mem_region {
             size: size as u64,
             guest_pfn: addr,
             userspace_addr: data as u64,
-            flags,
+            flags: flags,
+            rsvd: [0; 7],
```

Comment:
Note: the definition of mshv_user_mem_region has changed since this code was written, so I added the necessary fixes to it.